src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -1022,3 +1197,44 @@ bool InputStateMachineEngine::_GetXYPosition(const std::basic_string_view<size_t
 
     return success;
 }
+
+// Routine Description:
+// - Retrieves an X/Y coordinate pair for an SGR Mouse sequence from the parameter pool stored during Param actions.
+// Arguments:
+// - parameters - set of numeric parameters collected while pasring the sequence.
+// - line - Receives the Y/Line/Row position
+// - column - Receives the X/Column position
+// Return Value:
+// - True if we successfully pulled the cursor coordinates from the parameters we've stored. False otherwise.
+bool InputStateMachineEngine::_GetSGRXYPosition(const std::basic_string_view<size_t> parameters,
+                                                size_t& line,
+                                                size_t& column) const noexcept
+{
+    bool success = true;
+    line = DefaultLine;
+    column = DefaultColumn;
+
+    // SGR Mouse sequences have exactly 3 parameters
+    if (parameters.size() == 3)
+    {
+        column = til::at(parameters, 1);
+        line = til::at(parameters, 2);
+    }
+    else
+    {
+        success = false;

```

Comment:
This function too suffers from continuing to process when we've determined failure by not checking `success`. We should probably just early return this one too.